demo/api/src/importer/importers/product-importer/product-importer.ts

```diff
@@ -0,0 +1,54 @@
+import { EntityManager } from "@mikro-orm/core";
+import { Injectable, Logger, LoggerService } from "@nestjs/common";
+import { DataStream, DataStreamAndMetaData } from "@src/importer/data-streams/data-stream";
+import { pipeline, Transform } from "stream";
+
+@Injectable()
+export class ProductImporter {
+    dataStream: DataStreamAndMetaData | null = null;
+    name = "rawProductImport";
+    transformPipes: Transform[] = [];
+    logger: LoggerService;
+
+    constructor(private readonly em: EntityManager) {
+        this.logger = new Logger("product-importer");
+        this.transformPipes = [
+            new Transform({
+                objectMode: true,
+                transform: this.displayData.bind(this),
+            }),
+        ];
+    }
+
+    async init({ dataStream }: { dataStream: DataStream }): Promise<void> {
+        this.dataStream = await dataStream.getDataStreamsAndMetaData({ logger: this.logger });
+    }
+
+    getName(): string {
+        return this.name;
+    }
+
+    async executeRun(): Promise<boolean> {
+        if (this.dataStream) {
+            return new Promise((resolve, reject) => {
+                if (this.dataStream) {
```

Comment:
Why two ifs? 

I suppose it's got something to do with this.dataStream being outside the function scope. Wouldn't it make more sense to copy the dataStream to a local variable that's function scoped and use it inside the Promise?

Reply:
Yes that was the reason. I forgot this was still in there